src/main/java/mate/academy/dtoapp/repository/BookRepositoryImpl.java

```diff
@@ -1,4 +1,48 @@
 package mate.academy.dtoapp.repository;
 
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dtoapp.model.Book;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
 public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.createQuery("SELECT e FROM Book e", Book.class).getResultList();
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
```

Comment:
Should we throw an exception if book not exists?

Reply:
++, thank you! changed